src/app.ts

```diff
@@ -117,44 +106,154 @@ async function runWizard() {
         if (filename.includes("youtube.com")) {
           try {
             const transcript = await YoutubeTranscript.fetchTranscript(filename);
             wizardState.loadedPrimarySource = transcript.map((line) => line.text).join("\n");
-            wizardState.primarySourceFilename = filename;
-          } catch(err) {
+            wizardState.primarySourceFileOrFolderName = filename;
+          } catch (err) {
             return `Looked like a youtube video - Couldn't fetch transcript from ${filename}: ${err}`;
           }
-        } else if(!fs.existsSync(parsePlatformIndependentPath(filename))) {
-          return `File not found - tried to load ${filename}. Try again.`;
         } else {
-          try {
-            const dataFromFile = fs.readFileSync(parsePlatformIndependentPath(filename), "utf-8");
-            wizardState.loadedPrimarySource = dataFromFile;
-            wizardState.primarySourceFilename = parsePlatformIndependentPath(filename);
-          } catch(err) {
-            return `Couldn't read file - tried to load ${filename}. Try again.`;
+          const parsed_filename = parsePlatformIndependentPath(filename);
+          if (!fs.existsSync(parsed_filename)) return `File not found - tried to load ${filename}. Try again.`;
+
+          const file_stats = fs.lstatSync(parsed_filename);
+          if (file_stats.isFile()) {
+            if (!checkFileIsReadable(parsed_filename)) {
+              return `File type not supported - tried to load ${filename}. Try again.`;
+            } else {
+              try {
+                const dataFromFile = fs.readFileSync(parsePlatformIndependentPath(filename), "utf-8");
+                wizardState.loadedPrimarySource = dataFromFile;
+                wizardState.primarySourceFileOrFolderName = parsePlatformIndependentPath(filename);
+              } catch (err) {
+                return `Couldn't read file - tried to load ${filename}. Try again.`;
+              }
+            }
+          } else if (file_stats.isDirectory()) {
+            wizardState.primarySourceFileOrFolderName = parsed_filename;
+          } else if (!file_stats.isDirectory() && !file_stats.isFile()) {
+            return `Doesn't seem to be a file or a directory - tried to load ${filename}. Try again.`;
           }
         }
       }
-      return true;
+      return true;  // do we need to include this? I think 
     }
   });
-
   saveState(wizardState);
 
-  if(!wizardState.loadedPrimarySource) {
+
+  // ________________________________________FOLDER MANAGEMENT SECTION____________________________________
+
+  if (wizardState.primarySourceFileOrFolderName && fs.lstatSync(wizardState.primarySourceFileOrFolderName).isDirectory()) {
+    const parsed_filename = parsePlatformIndependentPath(wizardState.primarySourceFileOrFolderName);
+    let fileTree: dirTree.DirectoryTree | "timeoutFailed" = await getFileTree(parsed_filename);
```

Comment:
Does this have a potential to error out? Should we add a try catch?

Reply:
I don't think so @hrishioa ? If there's an error it will just return `timeoutFailed`, and I handled that. If you know of another error, can you add the handling? I haven't run into any